test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i

```diff
@@ -0,0 +1,82 @@
+[Mesh]
+  file = two_step_solve_master_cp/0002_mesh.cpr
+[]
+
+[Problem]
+  restart_file_base = two_step_solve_master_cp/LATEST
+  allow_sub_app_restart = true
```

Comment:
If you think it's cleaner to have the new flag to help out... then we can
do that.

You are right that this is complicated...
On Wed, Mar 9, 2016 at 1:25 PM Cody Permann <notifications@github.com>
wrote:

> In
> test/tests/restart/restart_subapp_not_master/two_step_solve_sub_restart.i
> <https://github.com/idaholab/moose/pull/6531#discussion_r55563732>:
>
> > @@ -0,0 +1,82 @@
> > +[Mesh]
> > +  file = two_step_solve_master_cp/0002_mesh.cpr
> > +[]
> > +
> > +[Problem]
> > +  restart_file_base = two_step_solve_master_cp/LATEST
> > +  allow_sub_app_restart = true
>
> Oh this is way more complicated than that. We do have a few restart tests
> that "appear" to work with recover (even though they don't actually #5877
> <https://github.com/idaholab/moose/issues/5877>). Some of this
> restart/recover logic gets very complicated in that scenario. I'm having a
> really difficult time pulling that flag out and getting all the right paths
> to work right now.
>
> It may be worth disallowing restart + recover altogether including
> disabling the currently broken tests instead of allowing them to pass
> incorrectly as I pull that flag out.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6531/files#r55563732>.
>